### DISCUSSION TITLE
Problem using global fetcher (SWRConfig component) in pages

### DISCUSSION THREAD

**Discussion by nora-kauczor** [neuefische/questions#835] [Web] [answered]
- Fetcher declared in [_app.js](https://github.com/Larissartoricader/Capstone/blob/feature/backend-read/pages/_app.js) and passed using `SWRConfig` cannot fetch data in [home page](https://github.com/Larissartoricader/Capstone/blob/feature/backend-read/pages/index.js) and [details page](https://github.com/Larissartoricader/Capstone/blob/feature/backend-read/pages/%5Bid%5D.js).
- Earlier attempts to fetch data directly in _app.js as well as fetching on a page with a fetcher declared just here (and not passed from above) were successful.
- I'm aware anyone trying to help can only search for our mistakes in theory, as you do not have our connection string. Please contact me if it might be easier just to have a look at the browser. Thank you.

**Comment by NikRoe** [comment, marked as answer]
Hi Nora,

the issue is most likely related to the return statement in your `_app.js`.

There you need to wrap the SWRConfig component around the `Component`. This will give all pages access to the `fetcher`.
It should look like this:

```js
return (
    <>
      <GlobalStyle />
      <SWRConfig value={{ fetcher }}>
      <Component
        {...pageProps}
        onAddRecipe={handleAddRecipe}
        onHandleBookmarkedIcon={handleBookmarkedIcon}
        bookmarkedRecipesIDs={bookmarkedRecipesIDs}
        onEditRecipe={handleEditRecipe}
      />
      </SWRConfig>
      <NavigationBar />
    </>
  );
```

I hope this already does the trick, let me know otherwise 🙃 ✌️

**Reply by nora-kauczor** [reply]
🤦🏼‍♀️🤦🏼‍♀️🤦🏼‍♀️🤦🏼‍♀️🤦🏼‍♀️🤦🏼‍♀️
Yes, that was the only problem. Thank you!

**Reply by NikRoe** [reply]
Awesome, I am glad that it works now 🥳
